### PR TITLE
:seedling: Rename golint to golangci-lint

### PR DESCRIPTION
**Pull request by ren-yifei**

### CODE REVIEW

**Review by kashifest — APPROVED** [review]

**Review by lentzi90 — COMMENTED** [review]
/lgtm
/hold
I think we need to wait for https://github.com/metal3-io/cluster-api-provider-metal3/pull/570 before we can merge this, right?

**Review by furkatgofurov7 — COMMENTED** [review]
/hold
You are renaming golint.sh script in release-0.5/release-0.4 branches, are there already PRs for those?

**Pull request merged**
